bundles/org.openhab.binding.solarman/src/main/java/org/openhab/binding/solarman/internal/channel/SolarmanChannelManager.java

```diff
@@ -0,0 +1,103 @@
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solarman.internal.channel;
+
+import static org.openhab.binding.solarman.internal.SolarmanBindingConstants.DYNAMIC_CHANNEL;
+import static org.openhab.binding.solarman.internal.typeprovider.ChannelUtils.escapeName;
+
+import java.math.BigDecimal;
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.solarman.internal.defmodel.InverterDefinition;
+import org.openhab.binding.solarman.internal.defmodel.ParameterItem;
+import org.openhab.binding.solarman.internal.typeprovider.ChannelUtils;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.type.ChannelKind;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Catalin Sanda - Initial contribution
+ */
+@NonNullByDefault
+public class SolarmanChannelManager {
+    private final ObjectMapper objectMapper;
+
+    public SolarmanChannelManager() {
+        objectMapper = new ObjectMapper();
+    }
+
+    public Map<ParameterItem, Channel> generateItemChannelMap(Thing thing, InverterDefinition inverterDefinition) {
+        return inverterDefinition.getParameters().stream().flatMap(parameter -> {
+            String groupName = escapeName(parameter.getGroup());
+
+            return parameter.getItems().stream().map(item -> {
+                String channelId = groupName + "-" + escapeName(item.getName());
+
+                Channel channel = ChannelBuilder.create(new ChannelUID(thing.getUID(), channelId))
+                        .withType(ChannelUtils.computeChannelTypeId(inverterDefinition.getInverterDefinitionId(),
+                                groupName, item.getName()))
+                        .withLabel(item.getName()).withKind(ChannelKind.STATE)
+                        .withAcceptedItemType(ChannelUtils.getItemType(item))
+                        .withProperties(Map.of(DYNAMIC_CHANNEL, Boolean.TRUE.toString()))
+                        .withConfiguration(buildConfigurationFromItem(item)).build();
+                return new AbstractMap.SimpleEntry<>(item, channel);
+            });
+        }).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private Configuration buildConfigurationFromItem(ParameterItem item) {
+        Configuration configuration = new Configuration();
+
+        BaseChannelConfig baseChannelConfig = new BaseChannelConfig();
+
+        @Nullable
+        BigDecimal offset = item.getOffset();
+        if (offset != null) {
+            baseChannelConfig.offset = offset;
+        }
+
+        @Nullable
```

Comment:
item.getScale() is already annotated with `@Nullable` so this should not be needed to annotate here again